repl/repl.go

```diff
@@ -78,29 +84,77 @@ func EvalString(what string) (res string, errs []string, formatted string) {
 	return
 }
 
-func Interactive(in io.Reader, out io.Writer, options Options) {
+func Interactive(options Options) int {
 	options.NilAndErr = true
 	s := eval.NewState()
 	macroState := object.NewMacroEnvironment()
 
-	scanner := bufio.NewScanner(in)
 	prev := ""
-	prompt := PROMPT
+
+	term, err := terminal.Open()
+	if err != nil {
+		return log.FErrf("Error creating readline: %v", err)
+	}
+	defer term.Close()
+	term.LoggerSetup()
+	term.SetPrompt(PROMPT)
+	options.Compact = true // because terminal doesn't do well will multi-line commands.
+	term.NewHistory(options.MaxHistory)
+	_ = term.SetHistoryFile(options.HistoryFile)
+	// Regular expression for "!nn" to run history command nn.
+	historyRegex := regexp.MustCompile(`^!(\d+)$`)
```

Comment:
Move this out or you will compute the regexp on each run

Also that's somehow the idea behind MustCompile

Otherwise use Compile and catch error

Reply:
there is only exactly 1 run